bot/src/main/java/com/community/tools/service/NotificationService.java

```diff
@@ -0,0 +1,121 @@
+package com.community.tools.service;
+
+import com.community.tools.discord.DiscordService;
+import com.community.tools.dto.UserForTaskStatusDto;
+import com.community.tools.model.TaskNameAndStatus;
+import com.community.tools.service.github.DiscordGithubService;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import net.dv8tion.jda.api.entities.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class NotificationService {
+
+  private static final String NOT_FOUND_DISCORD_USER_MESSAGE = "User with this discord name"
+      + " not found";
+
+  @Value("${text.channel}")
+  private String textChannelName;
+  @Autowired
+  private DiscordService discordService;
+  @Autowired
+  private DiscordGithubService discordGithubService;
+
+  /**
+   * Sends a pull request (PR) update notification to the Discord user associated with the given
+   * GitHub username.
+   *
+   * @param gitHubName            The GitHub username for which the PR update notification is
+   *                              intended.
+   * @param taskNameAndStatusList A list containing TaskNameAndStatus objects representing the
+   *                              updated task statuses.
+   * @throws IllegalArgumentException If the Discord user corresponding to the GitHub username is
+   *                                  not found.
+   */
+  public void sendPullRequestUpdateNotification(String gitHubName,
+      List<TaskNameAndStatus> taskNameAndStatusList) {
+    String discordName = discordGithubService.getDiscordName(gitHubName);
+    Optional<User> userOptional = discordService.getUserByName(discordName);
+    if (userOptional.isPresent()) {
+      discordService.sendMessageToConversation(textChannelName,
+          buildNotification(userOptional.get(), taskNameAndStatusList));
+    } else {
+      throw new IllegalArgumentException(NOT_FOUND_DISCORD_USER_MESSAGE);
```

Comment:
should add discord name to the thrown exception

Reply:
Name added